crates/context_aware_config/src/api/dimension/handlers.rs

```diff
@@ -74,6 +78,44 @@ async fn create(
         }
     };
 
+    let dimension_name = create_req.dimension.to_string();
+
+    let existing_dimension = match dimension_table
+        .filter(dimension.eq(&dimension_name))
+        .first::<Dimension>(&mut conn)
+        .optional()
+    {
+        Ok(dim) => dim,
+        Err(e) => {
+            log::warn!(
+                "Failed to fetch dimension {}: {:?}. Proceeding as if it does not exist.",
+                dimension_name,
+                e
+            );
+            None
+        }
+    };
+
+    let description = match &create_req.description {
+        Some(desc) if !desc.trim().is_empty() => desc.clone(),
+        _ => {
+            if let Some(existing_dim) = &existing_dimension {
+                existing_dim.description.clone()
+            } else {
+                // No description provided and no existing dimension
+                // Return an error
+                log::error!(
+                    "No description provided and no existing dimension for {}.",
+                    dimension_name
+                );
+                return Err(bad_argument!(
+                    "No description provided for the new dimension {}.",
+                    dimension_name
+                ));
+            }
+        }
```

Comment:
this is should ideally be in a transaction, lets separate out the create and update requests, its difficult to verify these changes
separating these out will make all these logics way simpler

in insert, description is mandatory => create entity with the request as is
in update, description is optional => if description is sent use it, else fetch from db and use the old description in the update query